### PR TITLE
Unit Testing - Mint function

### PR DESCRIPTION
**Pull request by axelizsak**
Issue #63 

Hey! I did the following work :

- For the **get_available_money_amount function** I performed the following tests: several successful purchases, getting the initial amount, large purchase and checking that the amount is equal to 0 is functional.

- Implementation of the **cancel_mint** function, which changes the value of **Mint_cancel** + simple test using **is_canceled**.

- Scarb fmt

Enjoy the review :)

### CODE REVIEW

File: src/components/minter/interface.cairo [hunk 1]
```diff
@@ -22,4 +22,5 @@ trait IMint<TContractState> {
         amount: u256
     );
     fn public_buy(ref self: TContractState, money_amount: u256, force: bool) -> Span<u256>;
+    fn cancel_mint(ref self: TContractState);
```
Comment (julienbrs) [inline src/components/minter/interface.cairo]: should take boolean param to open or close the mint

File: src/components/minter/mint.cairo [hunk 2]
```diff
@@ -82,6 +83,11 @@ mod MintComponent {
         cc_distributed: Span<u256>,
     }
 
+    #[derive(Drop, starknet::Event)]
+    struct MintCanceled {
+        time: u64
```
Comment (julienbrs) [inline src/components/minter/mint.cairo:89]: Should also emit the boolean param

File: tests/test_mint.cairo [hunk 3]
```diff
@@ -160,3 +162,83 @@ fn test_is_public_buy() {
...
+    let times: Span<u64> = get_mock_times();
+    let absorptions: Span<u64> = get_mock_absorptions();
+    // Setup the project with initial values
+    setup_project(project_address, 8000000000, times, absorptions,);
+
+    // Start testing environment setup
+    start_prank(CheatTarget::One(minter_address), owner_address);
+    start_prank(CheatTarget::One(erc20_address), owner_address);
+    start_prank(CheatTarget::One(minter_address), owner_address);
+    start_prank(CheatTarget::One(erc20_address), owner_address);
```
Comment (julienbrs) [inline tests/test_mint.cairo:180]:
```suggestion
```
You can remove it as u already did it before

File: tests/test_mint.cairo [hunk 4]
```diff
@@ -160,3 +162,83 @@ fn test_is_public_buy() {
...
+    // Ensure the public sale is open
+    let sale_open = minter.is_public_sale_open();
+    assert(sale_open == true, 'public sale not open');
+
+    // Default initial amount of money = 8000000000
+    let initial_money: u256 = 8000000000;
+
+    // Verify if the initial value is correct
+    let remaining_money = minter.get_available_money_amount();
+    assert(remaining_money == initial_money, 'remaning money wrong value');
```
Comment (julienbrs) [inline tests/test_mint.cairo]: typo 'remaining' instead of 'remaning'. Also edit the other same typo on that file

**Review by julienbrs — CHANGES_REQUESTED** [review]
Some little changes, but overall LGTM. Nice testing of `get_available_money_amount`!

**Comment by axelizsak** [comment]
Code changes made and **mint_canel** test improved.

**Review by julienbrs — CHANGES_REQUESTED** [review]
Perfect! Can you please in event emit the timestamp and the bool?

**Review by julienbrs — APPROVED** [review]

**Pull request merged**